web/api/v1/api.go

```diff
@@ -103,17 +103,21 @@ type API struct {
 	targetRetriever       targetRetriever
 	alertmanagerRetriever alertmanagerRetriever
 
-	now func() model.Time
+	now                  func() model.Time
+	retrieveConfigString configStringFunc
```

Comment:
Just call it `config` and make it a plain `func() string`. Not everything needs a dedicated type, especially if it's such a one-off thing.